src/main/java/uk/nhs/digital/nhsconnect/nhais/service/RecepProducerService.java

```diff
@@ -36,96 +41,102 @@ private String toEdifact(List<Segment> segments) {
-//
-//        recepMessageSegments.forEach(Segment::validate);
-//
-//        return recepMessageSegments;
-//    }
-//
-//    private InterchangeTrailer mapToRecepInterchangeTrailer(Interchange receivedInterchangeFromHa) {
-//        return new InterchangeTrailer(receivedInterchangeFromHa.getInterchangeTrailer().getNumberOfMessages());
-//    }
-//
-//    private MessageTrailer mapToRecepMessageTrailer(int refSegmentCount) {
-//        if (refSegmentCount < 2) {
-//            throw new IllegalStateException("There should be at least 2 ref segments. 1..1 ref interchange and 1..N ref message");
-//        }
-//        // there are 5 const segments and variable count of ref segments
-//        return new MessageTrailer(5 + refSegmentCount);
-//    }
-//
-//    private DateTimePeriod mapToRecepTranslationDateTime(Interchange receivedInterchangeFromHa) {
-//        return new DateTimePeriod(
-//            receivedInterchangeFromHa.getTranslationDateTime().getTimestamp(),
-//            receivedInterchangeFromHa.getTranslationDateTime().getTypeAndFormat());
-//    }
-//
-//    private RecepHeader mapToRecipInterchangeHeader(Interchange interchange) {
-//        var recepSender = interchange.getInterchangeHeader().getRecipient();
-//        var recepRecipient = interchange.getInterchangeHeader().getSender();
-//
-//        return new RecepHeader(recepSender, recepRecipient, interchange.getInterchangeHeader().getTranslationTime());
-//    }
-//
-//    private RecepMessageHeader mapToRecipMessageHeader(Interchange interchange) {
-//        return new RecepMessageHeader()
-//            .setSequenceNumber(interchange.getMessageHeader().getSequenceNumber());
-//    }
-//
-//    private RecepBeginningOfMessage mapToRecepBeginningOfMessage(Interchange interchange) {
-//        return new RecepBeginningOfMessage(interchange.getTranslationDateTime().getTimestamp());
-//    }
-//
-//    private RecepNameAndAddress mapToRecepNameAndAddress(Interchange interchange) {
-//        return new RecepNameAndAddress(interchange.getInterchangeHeader().getSender());
-//    }
-//
-//    private ReferenceMessageRecep mapToReferenceMessageRecep(Interchange interchange) {
-//        return new ReferenceMessageRecep(
-//            interchange.getMessageHeader().getSequenceNumber(), ReferenceMessageRecep.RecepCode.SUCCESS);
-//    }
-//
-//    private ReferenceInterchangeRecep mapToReferenceInterchangeRecep(Interchange interchange) {
-//        return new ReferenceInterchangeRecep(
-//            interchange.getInterchangeHeader().getSequenceNumber(),
-//            ReferenceInterchangeRecep.RecepCode.RECEIVED,
-//            interchange.getInterchangeTrailer().getNumberOfMessages());
-//    }
+    private List<Segment> mapEdifactToRecep(Interchange receivedInterchangeFromHa) throws EdifactValidationException {
```

Comment:
This method is too long and has to many `recep` inside - try to split it in smaller parts